### PR TITLE
dispatch event for 404 errors

### PR DESCRIPTION
**Pull request by karakayasemi**
## Description
Dispatch event from possible 404 error locations.

## Related Issue
https://github.com/owncloud/security/issues/3

## Motivation and Context
The goal is to be able to detect 404 errors
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

@DeepDiver1975 @Peter-Prochaska  please review.

### CODE REVIEW

File: lib/public/Http/ResourceNotFoundEvent.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ */
+
+namespace OCP\Http;
+
+use OCP\IRequest;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * @since 10.0.2
```
Comment (DeepDiver1975) [inline lib/public/Http/ResourceNotFoundEvent.php]: 10.0.3

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

**Comment by karakayasemi** [comment]
@DeepDiver1975 the version is corrected.

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 the version is corrected.

thx - I'd like to get a second voice - @PVince81 mind having a look? THX

**Comment by PVince81** [comment]
I guess `GenericEvent` would be too generic ?

If we're sending events for 404, should we also send some for 403, etc ? (if yes, separately)

**Comment by karakayasemi** [comment]
I can rename ResourceNotFoundEvent class name as HttpEvents and we can dispatch all the events http events by extending this class. Similar to following: https://github.com/owncloud/core/blob/master/lib/public/Comments/CommentsEvent.php . @PVince81 is it okay for you?

**Comment by PVince81** [comment]
@karakayasemi renaming sounds good

**Comment by karakayasemi** [comment]
@DeepDiver1975 @PVince81 please review. Thanks

**Review by PVince81 — APPROVED** [review]
:+1: from me

**Comment by DeepDiver1975** [comment]
backport ?

**Comment by karakayasemi** [comment]
Currently, only Security app will use this event. Security app min version probably will be 10.1, since #28206's milestone 10.1 and it was not backported. If we are planning to release security app for 9.1, we need to backport #28206 also, else IMHO no need to backport.

**Comment by phil-davis** [comment]
This is not in `stable10`
@karakayasemi @PVince81 @DeepDiver1975
Backport to `stable10` branch?

**Comment by karakayasemi** [comment]
As far as I know, there is nothing which using this event currently, it was added for Security app, but nothing implemented yet. I am not sure that we want to keep it.

**Comment by PVince81** [comment]
@karakayasemi I suggest reverting from master then as that would qualify as "API addition that we need to maintain and keep stable in the future" if we let it in stable10

**Comment by karakayasemi** [comment]
revert in here #34835

**Pull request merged**